src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+import { Box, Typography, useTheme, Link } from '@mui/material';
+import React, { useState } from 'react';
+
+import FacebookStatic from './assets/Facebook.svg?url';
+import GithubStatic from './assets/Github.svg?url';
+import GoogleStatic from './assets/Google.svg?url';
+import {
+  getPageContainerStyles,
+  getFormStyles,
+  getLogoStyles,
+  getInputStyles,
+  getButtonStyles,
+  haveAccountTextStyles,
+  loginLinkStyles,
+  signupWithTextStyles,
+  getSocialIconStyles,
+  legalTextStyles,
+  legalLinkStyles,
+} from './SignUpPage.styles';
+
+import type { SignUpPageProps } from './SignUpPage.types';
+
+const FacebookIcon: string | undefined = FacebookStatic ?? undefined;
+const GithubIcon: string | undefined = GithubStatic ?? undefined;
+const GoogleIcon: string | undefined = GoogleStatic ?? undefined;
+
+type LogoFallbackProps = {
+  orientation?: string;
+  width?: string | number;
+  height?: string | number;
+  alt?: string;
+};
+
+type FormInputFallbackProps = {
+  label?: string;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  errorMessage?: string;
+  sx?: Record<string, unknown>;
```

Comment:
The fallback FormInput type uses sx?: Record<string, unknown>; that's an imprecise type. For MUI sx props, prefer SxProps<Theme> (or the project's Sx type) so type checking works and consumers know the expected shape.

**Sugerencia:**
```typescript
Import and use SxProps from '@mui/system' or the MUI package: sx?: SxProps<Theme>.
```